### PR TITLE
Avoid AUTH by OpenId if the account is disabled

### LINKED ISSUE #6621: The user has been disabled, but can still authenticated by OpenId.
The OpenId module does not seem to handle the User.IsEnabled field.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #6621

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs [hunk 1]
```diff
@@ -69,6 +69,13 @@ public async Task<IUser> AuthenticateAsync(string userName, string password, Act
                 return null;
             }
 
+            if (!(user as User).IsEnabled)
+            {
+                reportError(string.Empty, S["The specified account is disabled, please contact an administrator."]);
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs:74]: Could it be the same error message as line 56, where and account is locker from SignInManager?

`reportError(string.Empty, S["The specified user is not allowed to sign in."]);`

Or change SignIngManager if we have a custom implementation.

Comment (sebastienros) [reply src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs:74]: Otherwise LGTM.

**Comment by hishamco** [comment]
Seb I edit the message directly from GitHub, coz my local repo was corrupted ;( If it's looks good :shipit:

File: src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs [hunk 2]
```diff
@@ -69,6 +69,13 @@ public async Task<IUser> AuthenticateAsync(string userName, string password, Act
                 return null;
             }
 
+            if (!(user as User).IsEnabled)
+            {
+                reportError(string.Empty, S["The specified user is not allowed to sign in."]);
```
Comment (deanmarcussen) [inline src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs:74]: Can you check the `AccountController` @hishamco 

There are two checks there, one for email should be confirmed, the other for account disabled. And the strings are different for both.

Ideally the behaviour should be the same

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs:74]: Do you mean this one https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs#L166 if it is that the previous message before last commit looks similar, but Seb suggested that in https://github.com/OrchardCMS/OrchardCore/pull/6637#discussion_r453959838

**Pull request merged**